src/Microsoft.AspNetCore.Mvc.Razor.Host/ViewComponentTagHelperDescriptorConventions.cs

```diff
@@ -0,0 +1,99 @@
+        /// </summary>
+        public static readonly string ViewComponentTagHelperPropertyHeader = "__Generated__";
+
+        /// <summary>
+        /// Each custom type name for a view component's tag helper representation will end with this footer.
+        /// </summary>
+        public static readonly string ViewComponentTagHelperPropertyFooter = "ViewComponentTagHelper";
+
+        /// <summary>
+        /// Verifies whether a <see cref="TagHelperDescriptor"/> represents a <see cref="ViewComponent"/>.  
+        /// </summary>
+        /// <param name="descriptor">The <see cref="TagHelperDescriptor"/> to check.</param>
+        /// <returns>Whether a <see cref="TagHelperDescriptor"/> represents a <see cref="ViewComponent"/>.</returns>
+        public static bool IsViewComponentDescriptor(TagHelperDescriptor descriptor)
+        {
+            return (descriptor.PropertyBag.ContainsKey(ViewComponentProperty)
+                && descriptor.PropertyBag.ContainsKey(ViewComponentTagHelperProperty));
+        }
+
+        /// <summary>
+        /// Retrieves the view component name from the tag helper property bag.
+        /// </summary>
+        /// <param name="descriptor">The <see cref="TagHelperDescriptor"/>.</param>
+        /// <returns>The short name of the <see cref="ViewComponent"/> represented by the <see cref="TagHelperDescriptor"/>,
+        /// or null if the tag helper does not represent a view component.</returns>
+        public static string GetViewComponentName(TagHelperDescriptor descriptor)
+        {
+            if (!IsViewComponentDescriptor(descriptor)) return null;
+
+            var viewComponentName = descriptor.PropertyBag[ViewComponentProperty];
+            return viewComponentName;
+        }
+
+        /// <summary>
+        /// Retrieves the view component tag helper name from the tag helper property bag.
+        /// </summary>
+        /// <param name="descriptor">The <see cref="TagHelperDescriptor"/>.</param>
+        /// <returns>The custom type name for a <see cref="ViewComponent"/> represented by the <see cref="TagHelperDescriptor"/>,
+        /// or null if the tag helper does not represent a view component.</returns>
+        public static string GetViewComponentTagHelperName(TagHelperDescriptor descriptor)
+        {
+            if (!IsViewComponentDescriptor(descriptor)) return null;
+
+            var viewComponentTagHelperName = descriptor.PropertyBag[ViewComponentTagHelperProperty];
+            return viewComponentTagHelperName;
+        }
+
+        /// <summary>
+        /// Retrieves the assembly name from a view component descriptor.
+        /// </summary>
+        /// <param name="descriptor">The <see cref="ViewComponentDescriptor"/>.</param>
+        /// <returns>The name of the assembly containing the <see cref="ViewComponentDescriptor"/>.</returns>
+        public static string GetAssemblyName(ViewComponentDescriptor descriptor) =>
```

Comment:
Could this be done in the calling code? Doesn't seem like you need a separate method for this.